### DISCUSSION TITLE
Serial Port API or bindings - low hanging fruit to allow the adoption of Odin, by the Maker, Electronics and Mecanotronics community.

### DISCUSSION THREAD

**Discussion by joaocarvalhoopen** [odin-lang/Odin#2730] [Ideas/Requests]
Hello,

I have searched for a serial API or a serial binding to an API for the 3 main Operating Systems that Odin support, Linux, Windows and Mac and Raspberry Pi, and have found that at the moment there is no Serial port over USB communications support in Odin.

This is really low hanging fruit, because the bindings of a lib for the 4 OS's would allow for all the Makers, all the Electrical Engineers, and all the Thinkeres, and Digital Artists to use Odin as a substitute for C, for C++, for Rust, or even for Python to connect to Embedded devices and make a PC program they use or make.
This devices that would be connected to the PC, could be, Arduinos, ESP32, general professional ARM or RISC V microcontrollers,  FPGA's or DSP's.
This would open a all class of new users to use Odin for as there main language "driver".

So I called it a really low hanging fruit opportunity.

Thank you very much,

Best regards,
João Carvalho

**Comment by Kelimion** [comment]
Welcome to Odin, and all, but you do realize that none of these Issues you are raising are Issues (bugs). They're better placed under Discussions, or discussed on our Discord.

**Comment by joaocarvalhoopen** [comment]
Thank you Kelimion,
I posted it under new features, but it comes out equal to issues.

Best regards,
João Carvalho
